src/Core/OrganizationFeatures/OrganizationSubscriptions/AddSecretsManagerSubscriptionCommand.cs

```diff
@@ -0,0 +1,154 @@
+﻿using Bit.Core.Entities;
+using Bit.Core.Enums;
+using Bit.Core.Exceptions;
+using Bit.Core.Models.StaticStore;
+using Bit.Core.OrganizationFeatures.OrganizationSubscriptions.Interface;
+using Bit.Core.Repositories;
+using Bit.Core.Services;
+using Bit.Core.Utilities;
+
+namespace Bit.Core.OrganizationFeatures.OrganizationSubscriptions;
+
+public class AddSecretsManagerSubscriptionCommand : IAddSecretsManagerSubscriptionCommand
+{
+    private readonly IPaymentService _paymentService;
+    private readonly IOrganizationRepository _organizationRepository;
+    private readonly IOrganizationUserRepository _organizationUserRepository;
+    private readonly IOrganizationService _organizationService;
+    public AddSecretsManagerSubscriptionCommand(
+        IPaymentService paymentService,
+        IOrganizationRepository organizationRepository,
+        IOrganizationUserRepository organizationUserRepository,
+        IOrganizationService organizationService)
+    {
+        _paymentService = paymentService;
+        _organizationRepository = organizationRepository;
+        _organizationUserRepository = organizationUserRepository;
+        _organizationService = organizationService;
+    }
+    public async Task<Organization> SignUpAsync(Organization organization, int additionalSeats,
+        int additionalServiceAccounts)
+    {
+        ValidateOrganization(organization);
+
+        var plan = StaticStore.SecretManagerPlans.FirstOrDefault(p => p.Type == organization.PlanType);
+
+        ValidateSecretsManagerPlan(plan, organization, additionalSeats, additionalServiceAccounts);
+
+        if (plan.Type != PlanType.Free)
+        {
+            await _paymentService.AddSecretsManagerToSubscription(organization, plan, additionalSeats, additionalServiceAccounts);
+        }
+
+        organization.SmSeats = additionalSeats;
+        organization.SmServiceAccounts = additionalServiceAccounts;
```

Comment:
Both these properties store totals, including the amount included in the plan. e.g.
`organization.SmSeats = plan.BaseSeats + additionalSeats`
`organization.SmServiceAccounts = plan.BaseServiceAccounts + additionalServiceAccounts`

Reply:
done